mm2src/coins/eth/eth_swap_v2.rs

```diff
@@ -0,0 +1,159 @@
+use super::eth::{addr_from_raw_pubkey, gas_limit, wei_from_big_decimal, EthCoin, EthCoinType, SignedEthTx,
+                 TAKER_SWAP_V2};
+use super::{SendTakerFundingArgs, Transaction, TransactionErr};
+use enum_derives::EnumFromStringify;
+use ethabi::Token;
+use ethcore_transaction::Action;
+use ethereum_types::{Address, U256};
+use futures::compat::Future01CompatExt;
+use std::convert::TryInto;
+
+struct TakerFundingArgs<'a> {
+    dex_fee: U256,
+    payment_amount: U256,
+    maker_address: Address,
+    taker_secret_hash: &'a [u8],
+    maker_secret_hash: &'a [u8],
+    funding_time_lock: u32,
+    payment_time_lock: u32,
+}
+
+impl EthCoin {
+    pub(crate) async fn send_taker_funding_impl(
+        &self,
+        args: SendTakerFundingArgs<'_>,
+    ) -> Result<SignedEthTx, TransactionErr> {
+        let taker_swap_v2_contract = self
+            .swap_v2_contracts
+            .as_ref()
+            .map(|contracts| contracts.taker_swap_v2_contract)
+            .ok_or_else(|| TransactionErr::Plain(ERRL!("Expected swap_v2_contracts to be Some, but found None")))?;
+        let dex_fee = try_tx_s!(wei_from_big_decimal(&args.dex_fee.fee_amount().into(), self.decimals));
+
+        let payment_amount = try_tx_s!(wei_from_big_decimal(
+            &(args.trading_amount + args.premium_amount),
+            self.decimals
+        ));
+        let maker_address = try_tx_s!(addr_from_raw_pubkey(args.maker_pub));
+
+        let funding_time_lock: u32 = try_tx_s!(args.funding_time_lock.try_into());
+        let payment_time_lock: u32 = try_tx_s!(args.payment_time_lock.try_into());
+        let funding_args = TakerFundingArgs {
+            dex_fee,
+            payment_amount,
+            maker_address,
+            taker_secret_hash: args.taker_secret_hash,
+            maker_secret_hash: args.maker_secret_hash,
+            funding_time_lock,
+            payment_time_lock,
+        };
+
+        match &self.coin_type {
+            EthCoinType::Eth => {
+                let data = try_tx_s!(self.prepare_taker_eth_funding_data(&funding_args).await);
+                self.sign_and_send_transaction(
+                    payment_amount,
+                    Action::Call(taker_swap_v2_contract),
+                    data,
+                    U256::from(gas_limit::ETH_PAYMENT),
```

Comment:
We should start using gas limits from the `gas_limit` parameter, similar to the below.
https://github.com/KomodoPlatform/komodo-defi-framework/blob/dab5bd996fe5620623aac5b2f3cd4016c3c27a20/mm2src/coins/eth.rs#L3730-L3731
We should also set constants for v2 swap functions gas limits c.c. @dimxy

Reply:
I used `self.gas_limit` in next pr https://github.com/KomodoPlatform/komodo-defi-framework/pull/2169/files#diff-b1bb9a17c62e6d79bb33d5b8a0db3954cc6b1763cbd95c02977023e39f0ea44cR83

I will add gas limit changes into this #2129 pr as well then

Reply:
I guess we need to add new gas params for contract v2 calls (as they have different gas consumption)

Reply:
added todo comment https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/7570df79cbb1ce29c0d24285e112726578fb4e47

Reply:
Todo comments can be lost (forgotten about), we should also have it in the issue checklist, the issue checklist will be checked before closing the issue in case we forgot something.

Reply:
got it, added todo point here https://github.com/KomodoPlatform/komodo-defi-framework/issues/1895#issue-1786470565